### PR TITLE
Intrinsic signal imaging

### PR DESCRIPTION
**Pull request by pauladkisson**
This PR adds intrinsic signal imaging, which is used to guide the ephys probe placement

### CODE REVIEW

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import DeepDict
+from neuroconv.tools import nwb_helpers
+
+
+class Schneider2024IntrinsicSignalOpticalImagingInterface(BaseDataInterface):
+    """Intrinsic signal optical imaging interface for schneider_2024 conversion"""
+
+    keywords = ["intrinsic signal optical imaging"]
```
Comment (h-mayorquin) [inline src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py]: Shouldn't this be a tuple?

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml [hunk 2]
```diff
@@ -87,3 +87,17 @@ Behavior:
 
 Sorting:
   units_description: Neural spikes will be sorted offline using Kilosort 2.5 and Phy2 software and manually curated to ensure precise spike time acquisition.
+
+IntrinsicSignalOpticalImaging:
+  Module:
+    name: intrinsic_signal_optical_imaging
+    description: For precise targeting of auditory cortex, intrinsic optical imaging (IOS) will be performed using a 2-photon microscope (Neurolabware). The skull is first bilaterally thinned over a region of interest (ROI) and made translucent. On experiment day, 680nm red light (ThorLabs) is used to image the ROI. Data is collected via MATLAB running custom suites for online and offline analyses.
+  Images:
+    name: images
```
Comment (h-mayorquin) [inline src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml:96]: This does not follow the naming convention for the types, right?

https://nwbinspector.readthedocs.io/en/dev/best_practices/general.html#naming-conventions

Is that intentional? I am missing something?

Same for the processed image

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml [hunk 3]
```diff
@@ -87,3 +87,17 @@ Behavior:
...
...ry cortex was identified through tonotopic mapping using a temporally periodic acoustic stimulation consisting of sequences of single pure tones of ascending or descending frequencies (20 tones per sequence, 2 to 40kHz exponentially separated, 75dB SPL, 50ms tone duration, 500ms onset-to-onset interval). Tone sequences were delivered from a free-field speaker 10cm above the mouse and repeated 40-60 times (0.1Hz). To compute tonotopic maps, the time course of each pixel was first highpass filtered using a moving average (10s window). Next, a Fourier transform was computed to extract the phase and the power of the frequency component at the frequency of acoustic stimulation (0.1 Hz). The phase indicates the sound frequency driving the response of a pixel, and the power indicates the strength of its response. To account for the hemodynamic delay and compute maps of absolute tonotopy, the response time to ascending sequence of tones was subtracted from the response time to the descending sequence. From these maps of absolute tonotopy, equally spaced iso-frequency contour lines were extracted, color-coded for sound frequency, and overlaid on top of the image of the blood vessel pattern.
```
Comment (h-mayorquin) [inline src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml:103]: Nice! Very complete!

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            "ProcessedImage": {
+                "properties": {
+                    "name": {"type": "string"},
+                    "description": {"type": "string"},
+                },
+            },
+        }
+        return metadata_schema
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict):
```
Comment (h-mayorquin) [inline src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py:60]: I like how thorough you are when adding metadata as a yaml file and then the whole schema. I usually just hard-code the metadata in the interface itself but I think your approach is better.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            nwbfile=nwbfile,
+            name=isoi_metadata["Module"]["name"],
+            description=isoi_metadata["Module"]["description"],
+        )
+        raw_image = GrayscaleImage(
+            name=isoi_metadata["RawImage"]["name"],
+            data=raw_image_array,
+            description=isoi_metadata["RawImage"]["description"],
+        )
+        processed_image = RGBImage(
```
Comment (h-mayorquin) [inline src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py:82]: For the color case, did you check that the images is indeed read as RBG and in the right orientation? (just display after writing it or with the neurosfit. I have gotten some surprises in that direction with

**Review by h-mayorquin — APPROVED** [review]
LGTM. 

Do you think it would be a good idea to add devices that you mention in the description? 

I wonder if there is value to it.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py (hunk 1 above)
Comment (pauladkisson) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py]: Yes, see https://github.com/catalystneuro/cookiecutter-my-lab-to-nwb-template/pull/29

**Comment by pauladkisson** [comment]
> Do you think it would be a good idea to add devices that you mention in the description?

Yes, thanks for the suggestion.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml (hunk 2 above)
Comment (pauladkisson) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml:96]: There is some open discussion on this: https://github.com/rly/ndx-pose/issues/31
And given all the considerations mentioned on that issue, I think snake_case is the more appropriate/intuitive choice for naming, so I'm going to stick with it for this conversion.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py (hunk 5 above)
Comment (pauladkisson) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py:82]: Yes, I checked this earlier it matches the image coming out of the jpg.

Comment (h-mayorquin) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py:82]: As long as it is in the expected orientation when someone reads it from nwb that should be fine.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml (hunk 2 above)
Comment (h-mayorquin) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml:96]: I don't have a horse in this race but maybe it is important for you to know (or remember?) that when we discussed this in the LBNL - CN meeting Ben was very strong against changing the best practice and we did not come to any conclusion about it but Ryan promised to write some modified best practices. Maybe something to bring up again.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py (hunk 5 above)
Comment (pauladkisson) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py:82]: Although the overlaid image is flipped left/right relative to the original raw_image...will make a note and ask.

File: src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml (hunk 2 above)
Comment (pauladkisson) [reply src/schneider_lab_to_nwb/schneider_2024/schneider_2024_metadata.yaml:96]: Good to know, thx.

**Pull request merged**